kernel/build-kernel.sh

```diff
@@ -0,0 +1,252 @@
+	kernel_path=$(cd "linux-${version}" && pwd)
+
+
+}
+
+get_default_kernel_config(){
+	local version="${1}"
+	#Remove extra 'v'
+	local version="${version#v}"
+
+	local hypervisor="$2"
+	local kernel_arch="$3"
+
+	[ -n "${version}" ] || die "kernel version not provided"
+	[ -n "${hypervisor}" ] || die "hypervisor not provided"
+	[ -n "${kernel_arch}" ] || die "kernel arch not provided"
+
+	major_version=$(echo "${version}" | cut -d. -f1)
+	minor_version=$(echo "${version}" | cut -d. -f2)
+	config="${default_kernel_config_dir}/${kernel_arch}_kata_${hypervisor}_${major_version}.${minor_version}.x"
+	[ -f "${config}" ] || die "failed to find default config ${config}"
+	echo "${config}"
+}
+
+get_config_version(){
+	config_version_file="${default_patches_dir}/../kata_config_version"
+	if [ -f "${config_version_file}" ];then
+		cat "${config_version_file}"
+	else
+		echo "unknown"
+	fi
+}
+
+main()
+{
+	while getopts "a:c:hk:t:p:v:" opt
+	do
+	case "$opt" in
+			a)
+				arch_target="${OPTARG}"
+				;;
+			c)
+				kernel_config_path="${OPTARG}"
+				;;
+
+			h)
+				usage
+				exit 0
+				;;
+
+			k)
+				kernel_path="${OPTARG}"
+				exit 0
+				;;
+
+			t)
+				hypervisor_target="${OPTARG}"
+				;;
+			p)
+				patches_path="${OPTARG}"
+				;;
+			v)
+				kernel_version="${OPTARG}"
+				;;
+			esac
+	done
+
+	shift $[$OPTIND-1]
+
+
+	if [ -z "$kernel_path" ]; then
+		info "kernel path not provided, will download kernel"
+		download_kernel="true"
+		# If not kernel version take it from versions.yaml
+		if [ -z "$kernel_version" ]; then
+			check_version_yaml && kernel_version=$(get_from_kata_deps "assets.kernel.version")
+		fi
+	fi
+	[ -n "$kernel_version" ] ||  die "failed to get kernel version: Kernel version is emtpy"
+
+	if [[ "${download_kernel}" == "true" ]];then
+		get_kernel "${kernel_version}"
+	fi
+
+	[ -n "$kernel_path" ] ||  die "failed to find kernel source path"
+
+	[ -n "${patches_path}" ] || patches_path="${default_patches_dir}"
+	[ -d "${patches_path}" ] ||  die " patches path '${patches_path}' does not exist"
+
+	kernel_patches=$(find "${patches_path}" -name '*.patch' -type f)
+
+	pushd "${kernel_path}" >> /dev/null
+	for p in ${kernel_patches};do
+		info "Applying patch $p"
+		patch -p1 < "$p"
+	done
+	[ -n "${hypervisor_target}" ] || hypervisor_target="kvm"
+	[ -n "${arch_target}" ] || arch_target="$(arch)"
+	[ -n "${kernel_config_path}" ] || kernel_config_path=$(get_default_kernel_config "${kernel_version}" "${hypervisor_target}" "${arch_target}")
+# 
+	config_version=$(get_config_version)
+
+	[ -n "${config_version}" ] || die "failed to get config version"
+	cp "${kernel_config_path}" ./.config
+	make oldconfig
+	make -j $(nproc)
+	install_path=$(readlink -m  "${DESTDIR}/${PREFIX}/share/${project_name}")
+	vmlinuz="vmlinuz-${kernel_version}-${config_version}"
+	vmlinux="vmlinux-${kernel_version}-${config_version}"
+	sudo install --mode 0644 -D "arch/${arch_target}/boot/bzImage" "${install_path}/${vmlinuz}"
```

Comment:
The condensed vmlinuz isn't saved with arch/arm64/boot/bzImage. Maybe we can create something like configure_<arch>.sh to store architecture-specific variant. @jcvenegas @jodh-intel